Source/FullScreenPokemon.ts

```diff
@@ -2511,6 +2524,8 @@ module FullScreenPokemon {
                     this.StateHolder.addChange(other.id, "sight", undefined);
                 }
             }
+
+            this.autoSave();
```

Comment:
This is for if the character being talked to isn't a trainer and didn't go into that block you mentioned so it there is a chance to save after dialog for those situations
Am I thinking about this wrong?